rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol

```diff
@@ -216,40 +221,47 @@ contract RBTCWrapperProxy is ContractRegistryClient {
     function addLiquidityToV1(
         address _liquidityPoolConverterAddress,
         IERC20Token[] memory _reserveTokens, 
-        uint256[] memory _reserveAmounts, 
+        uint256[] memory _reserveAmounts,
         uint256 _minReturn
     )  
         public
         payable
         checkAddress(_liquidityPoolConverterAddress)
         returns(uint256) 
     {
-        require(address(_reserveTokens[0]) == wrbtcTokenAddress, "The first reserve token must be WRBTC");
-        require(_reserveAmounts[0] == msg.value, "The provided amount of RBTC should be identical to msg.value");
+        require(address(_reserveTokens[1]) != wrbtcTokenAddress, "WRBTC must be in the 0 index");
 
         bool success;
-        uint256 amount;
         ILiquidityPoolV1Converter _liquidityPoolConverter = ILiquidityPoolV1Converter(_liquidityPoolConverterAddress);
         ISmartToken _poolToken = ISmartToken(address(_liquidityPoolConverter.token()));
         uint32 reserveRatio_ = _liquidityPoolConverter.reserveRatio();
         uint256 totalSupplyBefore = _poolToken.totalSupply();
         uint256[] memory rsvBalances = new uint256[](_reserveTokens.length);
-
-        IWrbtcERC20(wrbtcTokenAddress).deposit.value(_reserveAmounts[0])();
-        success = IWrbtcERC20(wrbtcTokenAddress).approve(_liquidityPoolConverterAddress, _reserveAmounts[0]);
-        require(success, "Failed to approve converter to transfer WRBTC");
         
-        for (uint256 i = 1; i < _reserveTokens.length; i++) {
-            success = IERC20Token(_reserveTokens[i]).transferFrom(msg.sender, address(this), _reserveAmounts[i]);
-            require(success, "Failed to transfer reserve token from user");
-            success = IERC20Token(_reserveTokens[i]).approve(_liquidityPoolConverterAddress, _reserveAmounts[i]);
-            require(success, "Failed to approve converter to transfer reserve token");
+        for (uint256 i = 0; i < _reserveTokens.length; i++) {
+            if(address(_reserveTokens[i]) == wrbtcTokenAddress) {
+                require(_reserveAmounts[i] == msg.value, "The provided amount of RBTC should be identical to msg.value");
+
+                IWrbtcERC20(wrbtcTokenAddress).deposit.value(_reserveAmounts[i])();
+                success = IWrbtcERC20(wrbtcTokenAddress).approve(_liquidityPoolConverterAddress, _reserveAmounts[i]);
+                require(success, "Failed to approve converter to transfer WRBTC");
+            } else {
+                if (i == 0) {
+                    require(_reserveAmounts[i] == 0, "0 amount required for non-wrbtc pairs");
+
+                    // Swap rbtc to reserve token --> Use convertion helper to avoid stack too deep
+                    _reserveAmounts[i] = convertionHelper(wrbtcTokenAddress, address(_reserveTokens[i]), address(this), msg.value);
```

Comment:
i think here we've got a grave misunderstanding. this makes no sense to me. in case that we have an ERC20/ERC20 pair, there is simply no RBTC involved.